### PR TITLE
Allow overwriting a SQLLab query that has previously been saved

### LINKED ISSUE #8068: Editing a saved query  in the sql_lab generate the new one instead of overwrite it!
When you want to edit a saved query with sql_lab and when you edit the query and click on the save query button in the sql_lab to save the query, superset makes a new query instead of overwriting the query!
So here is the bug! when I want to edit a saved query and make changes on it, I will get to the  Saved Queries pages and I will click on the pop-up link to route the sql_lab to modify the query instead of using the edit button on the saved query page. 
why should I or other users do this? here are the reasons:

1. Users create and develop the queries on the sql_lab page, so when a user wants
   to modify the query the user habit is to modify it on the sql_lab!
2. Users want to run and test the query when they are modifying it, so the edit page of the saved query is no absolutely sufficient!
3. sql_lab provides more functionality than the edit the record on the saved queries.
  
So my approach to solving this issue is that the saved query will be able to modify on the sql_lab, it's simple!

### Expected results

overwrite the edited query

### Actual results

Create the new record(query)

#### How to reproduce the bug

1. Go to the saved query page. 
2. Click on the pop-up link of any query that you want
3. modify the query and click on the save query button
4. See the saved queries page again.

### Environment

(please complete the following information):

- superset version: `superset version`
- python version: `python --version`
- node.js version: `node -v`
- npm version: `npm -v`

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by suddjian**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

When saving a query in SQLLab, currently there is no good way to update a query you were previously working on. You have to save a new one, then go to the saved queries list and delete the old one. This change lets you choose to overwrite the previous entry or save a new copy.

A speed bump I ran into was that the frontend is using several field names that are different from the names the backend uses. My first instinct was to change that so that a field means the same thing everywhere, but that gets pretty complicated pretty fast, due to the way the redux state is being persisted locally - how does the Superset frontend do migrations? #7748 will probably allow solving this problem.

So instead I grouped the existing differences (plus my addition of a `remoteId` field) into a `queryClientMapping` object, with the goal of isolating the differences to one layer of the system - where the client actually makes API calls. That way UI components like `SaveQuery` don't have to concern themselves with those differences. This hasn't been propagated to all parts of the code that use saved queries, yet.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
Screen Shot 2019-09-25 at 7 04 01 PM
After:
Screen Shot 2019-09-25 at 7 03 30 PM

### TEST PLAN

I've already tested it pretty well, I think. This is my first contribution to Superset so it's entirely possible that I missed something. It would be nice to get some additional verification that nothing I've done will mess up anyone's local state.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #8068
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/sql_lab.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ class SavedQueryViewApi(SavedQueryView):
         "sql",
         "extra_json",
     ]
-    show_columns = ["label", "db_id", "schema", "description", "sql", "extra_json"]
+    show_columns = ["id", "label", "db_id", "schema", "description", "sql", "extra_json"]
```
Comment (mistercrunch) [inline superset/views/sql_lab.py]: I think `black` thinks this line is too long, enabling the pre-commit hooks should help prevent this type of things
https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md#git-hooks

**Review by mistercrunch — APPROVED** [review]
Minor comments and some red on the builds, otherwise LGTM

**Comment by suddjian** [comment]
There's one cypress test that failed in that last build, but I can't reproduce it on my machine. It's probably flaky. If someone with the necessary auth could re-run that job it'd be much appreciated.

**Comment by villebro** [comment]
I restarted the build, I believe this is due to flakiness.

**Comment by graceguo-supercat** [comment]
@suddjian this PR broke existed save query feature. when i try to save a query from sql lab, i got server-side error message:

Screen Shot 2019-10-03 at 9 33 08 PM

**Comment by suddjian** [comment]
Thanks for the report @graceguo-supercat I could have sworn I saw this working! I am able to repro it though, so hopefully I'll have a fix up today.

**Pull request merged**